consensus/src/genesis/genesis_init.rs

```diff
@@ -0,0 +1,24 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0
+
+use mina_rs_base::types::ExternalTransition;
+
+use super::*;
+
+///
+/// # Example
+/// ```rust
+/// use mina_rs_base::types::*;
+/// use mina_consensus::genesis::{GenesisInit, DEVNET_CONFIG, MAINNET_CONFIG};
+/// let genesis_mainnet = ExternalTransition::init_genesis(&MAINNET_CONFIG);
+/// let genesis_devnet = ExternalTransition::init_genesis(&DEVNET_CONFIG);
+/// ```
+pub trait GenesisInit {
+    fn init_genesis(config: &GenesisInitConfig) -> ExternalTransition;
```

Comment:
I wonder if we could do something like:

```
trait GenesisInit {
    const CONFIG: Config = config::load()
}
```
config::load would load from conditionally compiled modules.
what do you think?

Reply:
That will be great but I doubt everything can fit into const fn (e.g. PublicKeys)

Reply:
right